Test regression: Failed to set SO_LINGER

With srt-1.4.2 and 1.4.3 the test for setting SO_LINGER fails on one of my machines:

```go test .
--- FAIL: TestNewSocketLinger (0.00s)
    srtgo_test.go:34: Could not create a srt socket
FAIL
FAIL	github.com/haivision/srtgo	0.007s
FAIL
```

Looking at the code it is using the native syscall.SetsockoptLinger directly on the srt socket which fails. Using the commented out version with srt_setsockopt fixes this for me, however I haven't yet tested whether the option is actually set.

https://github.com/Haivision/srtgo/blob/27de7e427495c78b590dc782943048e44abe4710/srtsocketoptions.go#L103-L113

--- a/srtgo_test.go
+++ b/srtgo_test.go
@@ -31,7 +31,16 @@ func TestNewSocketLinger(t *testing.T) {
 	a := NewSrtSocket("localhost", 8090, options)
 
 	if a == nil {
-		t.Error("Could not create a srt socket")
+		t.Error("Could not create a srt socket with linger")
+	}
+
+	// read back value to make sure
+	res, err := getSocketLingerOption(a)
+	if err != nil {
+		t.Error(err)
+	}
+	if res != 1000 {
+		t.Error("Failed to set linger option")
 	}
 }
 

--- a/srtsocketoptions.go
+++ b/srtsocketoptions.go
@@ -5,6 +5,7 @@ package srtgo
 import "C"
 
 import (
+	"errors"
 	"fmt"
 	"strconv"
 	"syscall"
@@ -108,8 +109,24 @@ func setSocketLingerOption(s C.int, li int32) error {
 	} else {
 		lin.Onoff = 0
 	}
-	// C.srt_setsockopt(s.socket, bindingPre, C.SRTO_LINGER, unsafe.Pointer(&lin), C.int(unsafe.Sizeof(lin)))
-	return syscall.SetsockoptLinger(int(s), bindingPre, syscall.SO_LINGER, &lin)
+	res := C.srt_setsockopt(s, bindingPre, C.SRTO_LINGER, unsafe.Pointer(&lin), C.int(unsafe.Sizeof(lin)))
+	if res == SRT_ERROR {
+		return errors.New("failed to set linger")
+	}
+	return nil
+}
+
+func getSocketLingerOption(s *SrtSocket) (int32, error) {
+	var lin syscall.Linger
+	size := int(unsafe.Sizeof(lin))
+	err := s.getSockOpt(C.SRTO_LINGER, unsafe.Pointer(&lin), &size)
+	if err != nil {
+		return 0, err
+	}
+	if lin.Onoff == 0 {
+		return 0, nil
+	}
+	return lin.Linger, nil
 }
 
 // Set socket options for SRT

